File: CS/DXWindowsApplication5/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraGrid.Columns;


namespace DXWindowsApplication5
{
    public partial class Form1 : XtraForm
    {
        public Form1()
        {
            InitializeComponent();

            AddGridView();
            RegisterViewsInComboBox();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'countriesDBDataSet.Countries' table. You can move, or remove it, as needed.
            this.countriesTableAdapter.Fill(this.countriesDBDataSet.Countries);
         }

        GridView myGridView;

        private void AddGridView()
        {
            myGridView = new GridView(gridControl1);
            myGridView.ViewCaption = "Table";


            string[] fieldNames = new string[] { "Country", "Capital", 
                "Currency", "Religions", "Area", "Population", "GDP", "Languages" };

            foreach (string fieldName in fieldNames) {
                GridColumn column = myGridView.Columns.AddField(fieldName);
                column.Visible = true;
            }
        }

        private void RegisterViewsInComboBox()
        {
            comboView.Items.Add("Auto");
            foreach (BaseView view in gridControl1.ViewCollection)
                comboView.Items.Add(view.ViewCaption);
            comboViewTypeItem.EditValue = gridControl1.ViewCollection[0].ViewCaption;
        }

        private void barComboItem_EditValueChanged(object sender, EventArgs e)
        {
            ComboBoxEdit comboBox = barManager.ActiveEditor as ComboBoxEdit;
            if (comboBox == null) 
                return;
            if (comboBox.EditValue.ToString() == "Auto")
                SetAutoView();
            else
            {
                BaseView selectedView = gridControl1.ViewCollection[comboBox.SelectedIndex - 1];
                gridControl1.MainView = selectedView;
            }
        }

        private void gridControl1_Resize(object sender, EventArgs e)
        {
            if (comboViewTypeItem.EditValue.ToString() != "Auto")
                return;
            SetAutoView();
        }

        private void SetAutoView()
        {
            if (gridControl1.Width < 700)
                gridControl1.MainView = layoutView1;
            else
                gridControl1.MainView = myGridView;
        }

        private void gridControl1_DefaultViewChanged(object sender, EventArgs e)
        {
            checkShowFooter.Enabled = gridControl1.MainView is GridView;
            if (checkShowFooter.Enabled)
                checkShowFooter.Checked = (gridControl1.MainView as GridView).OptionsView.ShowFooter;
            else
                checkShowFooter.Checked = false;
        }

        private void checkShowFooter_CheckedChanged(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            if (gridControl1.MainView is GridView)
                (gridControl1.MainView as GridView).OptionsView.ShowFooter = checkShowFooter.Checked;
        }
    }
}

File: VB/DXWindowsApplication5/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.XtraGrid.Columns


Namespace DXWindowsApplication5
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()

			AddGridView()
			RegisterViewsInComboBox()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'countriesDBDataSet.Countries' table. You can move, or remove it, as needed.
			Me.countriesTableAdapter.Fill(Me.countriesDBDataSet.Countries)
		End Sub

		Private myGridView As GridView

		Private Sub AddGridView()
			myGridView = New GridView(gridControl1)
			myGridView.ViewCaption = "Table"


			Dim fieldNames() As String = { "Country", "Capital", "Currency", "Religions", "Area", "Population", "GDP", "Languages" }

			For Each fieldName As String In fieldNames
				Dim column As GridColumn = myGridView.Columns.AddField(fieldName)
				column.Visible = True
			Next fieldName
		End Sub

		Private Sub RegisterViewsInComboBox()
			comboView.Items.Add("Auto")
			For Each view As BaseView In gridControl1.ViewCollection
				comboView.Items.Add(view.ViewCaption)
			Next view
			comboViewTypeItem.EditValue = gridControl1.ViewCollection(0).ViewCaption
		End Sub

		Private Sub barComboItem_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboViewTypeItem.EditValueChanged
			Dim comboBox As ComboBoxEdit = TryCast(barManager.ActiveEditor, ComboBoxEdit)
			If comboBox Is Nothing Then
				Return
			End If
			If comboBox.EditValue.ToString() = "Auto" Then
				SetAutoView()
			Else
				Dim selectedView As BaseView = gridControl1.ViewCollection(comboBox.SelectedIndex - 1)
				gridControl1.MainView = selectedView
			End If
		End Sub

		Private Sub gridControl1_Resize(ByVal sender As Object, ByVal e As EventArgs) Handles gridControl1.Resize
			If comboViewTypeItem.EditValue.ToString() <> "Auto" Then
				Return
			End If
			SetAutoView()
		End Sub

		Private Sub SetAutoView()
			If gridControl1.Width < 700 Then
				gridControl1.MainView = layoutView1
			Else
				gridControl1.MainView = myGridView
			End If
		End Sub

		Private Sub gridControl1_DefaultViewChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gridControl1.DefaultViewChanged
			checkShowFooter.Enabled = TypeOf gridControl1.MainView Is GridView
			If checkShowFooter.Enabled Then
				checkShowFooter.Checked = (TryCast(gridControl1.MainView, GridView)).OptionsView.ShowFooter
			Else
				checkShowFooter.Checked = False
			End If
		End Sub

		Private Sub checkShowFooter_CheckedChanged(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles checkShowFooter.CheckedChanged
			If TypeOf gridControl1.MainView Is GridView Then
				TryCast(gridControl1.MainView, GridView).OptionsView.ShowFooter = checkShowFooter.Checked
			End If
		End Sub
	End Class
End Namespace